### ISSUE TITLE
Question: How are you converting VP8

### ISSUE THREAD

**Issue by Terricide** [sipsorcery-org/VP8.Net#1]
I want to help out with this if I can. Are you using a tool to convert c++ to c# or you doing each file by hand? I'm interested in the getting the encoder working.

**Comment by sipsorcery** [comment]
By hand I'm afraid. There are lots of macros, unions and blocks of memory that get accessed interchangeably with pointer and arrays. The conversion tools I tried just created a separate set of problems to decipher.

The decoder took a month...

If that hasn't turned you off I can create a repo with the C source code I'm working from. It is the source from the libvpx repo but with the configure options tweaked to remove the platform specific assembly code.

**Comment by Terricide** [comment]
Sounds interesting, I wouldn't mind taking a peek at the source code and see what I can do to help.

**Comment by sipsorcery** [comment]
I've added the [libvpx C solution](https://github.com/sipsorcery/VP8.Net/tree/master/libvpx/build-win-x64) to the repo.

The approach I used with the decoder is to make the C# code as line-by-line identical with the C code as possible. This means it's possible to then debug line-by-line comparing the C and C# implementations to track down why the C# code doesn't work, which it never does on the first 50 attempts.

Optimising the C# code and using more suitable C# approaches would be stage 2.
